cirq-google/cirq_google/engine/engine.py

```diff
@@ -312,20 +312,35 @@ async def run_sweep_async(
         Raises:
             ValueError: If no gate set is provided.
         """
-        engine_program = await self.create_program_async(
-            program, program_id, description=program_description, labels=program_labels
-        )
-        return await engine_program.run_sweep_async(
+        if not program_id:
+            program_id = _make_random_id('prog-')
+        if not job_id:
+            job_id = _make_random_id('job-')
+        run_context = self.context._serialize_run_context(params, repetitions)
+
+        stream_job_response_future = self.context.client.run_job_over_stream(
+            project_id=self.project_id,
+            program_id=program_id,
+            program_description=program_description,
+            program_labels=program_labels,
+            code=self.context._serialize_program(program),
             job_id=job_id,
-            params=params,
-            repetitions=repetitions,
             processor_ids=processor_ids,
-            description=job_description,
-            labels=job_labels,
+            run_context=run_context,
+            job_description=job_description,
+            job_labels=job_labels,
+        )
+        return engine_job.EngineJob(
+            self.project_id,
+            program_id,
+            job_id,
+            self.context,
+            stream_job_response_future=stream_job_response_future,
         )
 
     run_sweep = duet.sync(run_sweep_async)
 
+    # TODO update
```

Comment:
I would either explain what needs to be updated, connect it to a github issue, or just delete the comment.
As is, it does not give much useful information.

Same for below.

Reply:
Will do - I left these here as a reminder for myself to implement this in the full PR.